engineering/typescript-interview/index.ts

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+import ProductApi from './src/Api';
+
+const client = axios.create({
+    baseURL: 'https://eve.theiconic.com.au',
+    timeout: 1000,
+});
+
+async function run(): Promise<void> {    
+    const productApi = new ProductApi(client);
+
+    /*
+        Your Challenge Starts here
+        Write a script that consumes our product API and decorates results with video preview links 
```

Comment:
#### **question** 🧐❓  


Would you mind adding a more detail about the end goal, for example:

- Base on X inputs import 10k elements, how do you track progress?
- Write in 10 mins
- With tests?

<sub> I'm using [conventionalcomments.org](https://conventionalcomments.org) to make the expectations of the review easier </sub>

Reply:
Good question, this will be covered in the corollary documentation for the interviewer.